x-pack/plugins/lens/public/metric_visualization/expression.tsx

```diff
@@ -70,18 +126,20 @@ export function MetricChart({
   if (!column || !row) {
     return getEmptyState();
   }
+  const rawValue = row[accessor] as number | undefined;
 
   // NOTE: Cardinality and Sum never receives "null" as value, but always 0, even for empty dataset.
   // Mind falsy values here as 0!
-  const shouldShowResults = row[accessor] != null;
-  if (!shouldShowResults) {
+  if (rawValue == null) {
```

Comment:
```suggestion
  const rawValue = row[accessor];

  // NOTE: Cardinality and Sum never receives "null" as value, but always 0, even for empty dataset.
  // Mind falsy values here as 0!
  if (typeof rawValue !== 'number') {
    return getEmptyState();
  }
```

This would allow us to avoid type assertion, WDYT?